### PR TITLE
Added support for handlers that return promises

### PR DESCRIPTION
**Pull request by jkehres**
With Node.js 8.10, your lambda handler can now directly return a promise instead of invoking `callback()`. `context.fail()`, `context.succeed()` or `context.done()`.

https://aws.amazon.com/blogs/compute/node-js-8-10-runtime-now-available-in-aws-lambda/

This PR adds two new expect methods, `expectResolve()` and `expectReject()`, to support handlers that return promises.

### CODE REVIEW

File: docs/promise.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+# Verifying Promises
```
Comment (jkehres) [inline docs/promise.md]: Copied these docs from `callback.md` and modified accordingly.

File: lib/index.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-const semver = require('semver')
+const semver = require('semver');
```
Comment (jkehres) [inline lib/index.js]: Unrelated fix.

File: test/lib/runner.test.js [hunk 3]
```diff
@@ -138,7 +138,64 @@ describe( 'lib/runner', function() {
 
                         expect( context.getRemainingTimeInMillis() ).to.be.above( 0 );
 
-                        callback( new Error( 'bang') );
+                        callback( new Error( 'bang' ) );
```
Comment (jkehres) [inline test/lib/runner.test.js]: Unrelated fix - spacing.

**Comment by jamesdixon** [comment]
@jkehres nice work!

I tried your fork and tests are now working properly. The only issue I'm having is that the test runner never exits after the tests complete. Have you run into this?

**Comment by HajoAhoMantila** [comment]
I created issue https://github.com/vandium-io/lambda-tester/issues/44 for exactly this feature before I noticed there's already this PR. Nice to see somebody already solved this!

There's also another PR (https://github.com/vandium-io/lambda-tester/pull/42) now that addresses the same thing. Instead of introducing additional promise-specific verifiers functions expectResolve/expectReject, the solution in that PR would reuse the existing expectResult/expectError  functions. As the callback and promise approaches are largely equivalent, this could keep the API of lambda-tester simpler. Any thoughts on this?

**Comment by mhnguyen** [comment]
Thanks @jkehres !
